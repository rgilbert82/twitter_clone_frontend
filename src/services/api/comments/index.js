export { default as getCommentsAPI } from './getComments.js';
export { default as createCommentAPI } from './createComment.js';
export { default as editCommentAPI } from './editComment.js';
export { default as deleteCommentAPI } from './deleteComment.js';
