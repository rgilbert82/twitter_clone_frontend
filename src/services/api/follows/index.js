export { default as getFollowAPI } from './getFollow.js';
export { default as createFollowAPI } from './createFollow.js';
export { default as deleteFollowAPI } from './deleteFollow.js';
export { default as getUserFollowersAPI } from './getUserFollowers.js';
export { default as getUserFollowingsAPI } from './getUserFollowings.js';
