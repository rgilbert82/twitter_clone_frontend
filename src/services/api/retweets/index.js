export { default as createRetweetAPI } from './createRetweet.js';
export { default as deleteRetweetAPI } from './deleteRetweet.js';
