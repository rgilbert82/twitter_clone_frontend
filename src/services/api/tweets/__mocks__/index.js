export { default as getTweetAPI } from './getTweet.js';
export { default as createTweetAPI } from './createTweet.js';
export { default as editTweetAPI } from './editTweet.js';
export { default as deleteTweetAPI } from './deleteTweet.js';
