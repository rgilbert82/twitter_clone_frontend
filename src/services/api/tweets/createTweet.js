import { get_API_URL } from '../../misc';

export default (tweetData) => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/tweets`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(JSON.parse(request.responseText));
        } else {
          reject(request.responseText);
        }
      }
    };

    request.open('POST', path);
    request.withCredentials = true;
    request.setRequestHeader('Authorization', document.cookie || '');
    request.setRequestHeader('Content-Type', 'application/json');
    request.send(JSON.stringify(tweetData));
  });
}
