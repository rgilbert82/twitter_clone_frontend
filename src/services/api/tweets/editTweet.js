import { get_API_URL } from '../../misc';

export default (tweetData) => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/tweets/${tweetData.id}`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(JSON.parse(request.responseText));
        } else {
          reject(request.responseText);
        }
      }
    }

    request.open('PUT', path);
    request.withCredentials = true;
    request.setRequestHeader('Authorization', document.cookie || '');
    request.setRequestHeader('Content-Type', 'application/json');
    request.send(JSON.stringify(tweetData));
  });
}
