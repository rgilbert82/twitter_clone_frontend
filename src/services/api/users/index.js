export { default as getUsersAPI } from './getUsers.js';
export { default as getUserAPI } from './getUser.js';
export { default as createUserAPI } from './createUser.js';
export { default as updateUserAPI } from './updateUser.js';
export { default as getUserTweetsAPI } from './getUserTweets.js';
export { default as deleteUserAPI } from './deleteUser.js';
