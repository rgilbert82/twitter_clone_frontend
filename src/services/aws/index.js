export { default as uploadPhoto } from './uploadPhoto.js';
export { default as deletePhoto } from './deletePhoto.js';
