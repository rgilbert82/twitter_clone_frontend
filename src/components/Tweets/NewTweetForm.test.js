jest.mock('../../services/api/tweets');
jest.mock('../../services/aws');

import React from 'react';
import { shallow } from 'enzyme';
import { NewTweetForm } from '.';
import { createTweetAPI } from '../../services/api/tweets';
import { uploadPhoto } from '../../services/aws';

describe(NewTweetForm, () => {
  const currentUser = { id: 1, name: 'Ryan', username: 'RG', token: '12345', slug: 'rg', followers: 1, following: 1, tweet_count: 1, retweet_count: 1, all_tweet_count: 2 };
  const toggleMock = jest.fn();
  let createSpy = jest.spyOn(NewTweetForm.prototype, 'uploadPhotoBeforeCreatingTweet');
  let component;

  beforeEach(() => {
    component = shallow(<NewTweetForm currentUser={currentUser} toggleTweetForm={toggleMock} displayMessage={jest.fn()} />);
  });

  afterAll(() => {
    createSpy.mockClear();
  });

  it("renders properly", () => {
    expect(component).toMatchSnapshot();
  });

  it("toggles the form", () => {
    component.find('#toggleForm').simulate('click');
    expect(toggleMock).toBeCalled();
  });

  it("updates the body", () => {
    expect(component.state('tweetBody')).toEqual('');
    component.find('#form_body').simulate('change', { target: { value: 'Hello' }});
    expect(component.state('tweetBody')).toEqual('Hello');
  });

  it("is a valid form", () => {
    expect(component.instance().isValidForm()).toEqual(false);
    component.find('#form_body').simulate('change', { target: { value: 'Hello' }});
    expect(component.instance().isValidForm()).toEqual(true);
  });

  it("calls the createTweet function", () => {
    component.find('#form_body').simulate('change', { target: { value: 'Hello' }});
    component.find('#submitForm').simulate('click');
    expect(createSpy).toHaveBeenCalled();
  });

  it("creates the tweet", () => {
    expect.assertions(1);
    component.setState({ tweetBody: 'Hello' });
    component.instance().createTweet()
      .then(() => {
        expect(toggleMock).toBeCalled();
      });
  });
});
