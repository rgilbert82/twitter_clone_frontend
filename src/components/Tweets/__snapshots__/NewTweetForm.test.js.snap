// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class NewTweetForm extends _react2.default.Component {
  constructor(props) {
    super(props);
    this.state = { tweetCreated: false, tweetBody: '' };
    this.updateBody = this.updateBody.bind(this);
    this.isValidForm = this.isValidForm.bind(this);
    this.createTweet = this.createTweet.bind(this);
    this.uploadPhotoBeforeCreatingTweet = this.uploadPhotoBeforeCreatingTweet.bind(this);
  }

  updateBody(e) {
    this.setState({ tweetBody: e.target.value });
  }

  isValidForm() {
    return this.state.tweetBody.trim().length > 0;
  }

  createTweet(photoPath) {
    return (0, _tweets.createTweetAPI)({ body: this.state.tweetBody, user_id: this.props.currentUser.id, image: photoPath }).
    then(() => {
      this.props.toggleTweetForm();
      this.props.raiseSidebarTweetCount();
    }).catch(() => {
      this.props.displayMessage('There was an error creating the tweet');
    });
  }

  uploadPhotoBeforeCreatingTweet() {
    return (0, _aws.uploadPhoto)(this.props.currentUser.id).
    then(photoPath => {
      this.createTweet(photoPath);
    }).catch(err => {
      this.props.displayMessage('There was an error uploading your image');
    });
  }

  render() {
    let component;

    if (this.state.tweetCreated) {
      component = _react2.default.createElement(_reactRouterDom.Redirect, { to: '/', __source: { fileName: _jsxFileName, lineNumber: 49 }, __self: this });
    } else {
      component =
      _react2.default.createElement('div', { id: 'modal_wrapper', __source: { fileName: _jsxFileName, lineNumber: 52 }, __self: this },
        _react2.default.createElement(_Misc.DarkLayer, { closeModal: this.props.toggleTweetForm, __source: { fileName: _jsxFileName, lineNumber: 53 }, __self: this }),
        _react2.default.createElement('div', { id: 'new_tweet_form', __source: { fileName: _jsxFileName, lineNumber: 54 }, __self: this },
          _react2.default.createElement('div', { className: 'modal_header', __source: { fileName: _jsxFileName, lineNumber: 55 }, __self: this },
            _react2.default.createElement('h1', { __source: { fileName: _jsxFileName, lineNumber: 56 }, __self: this }, 'Compose new Tweet')),

          _react2.default.createElement('div', { className: 'modal_body', __source: { fileName: _jsxFileName, lineNumber: 58 }, __self: this },
            _react2.default.createElement(_reactBootstrap.FormGroup, { __source: { fileName: _jsxFileName, lineNumber: 59 }, __self: this },
              _react2.default.createElement(_reactBootstrap.FormControl, { componentClass: 'textarea', id: 'form_body', maxLength: '140', value: this.state.tweetBody, onChange: this.updateBody, placeholder: 'What\\'s happening?', __source: { fileName: _jsxFileName, lineNumber: 60 }, __self: this })),

            _react2.default.createElement(_reactBootstrap.FormGroup, { __source: { fileName: _jsxFileName, lineNumber: 62 }, __self: this },
              _react2.default.createElement(_reactBootstrap.ControlLabel, { htmlFor: 'photo_upload', __source: { fileName: _jsxFileName, lineNumber: 63 }, __self: this }, 'Add an image'),
              _react2.default.createElement(_reactBootstrap.FormControl, { type: 'file', id: 'photo_upload', __source: { fileName: _jsxFileName, lineNumber: 64 }, __self: this })),

            _react2.default.createElement(_reactBootstrap.Button, { id: 'toggleForm', onClick: this.props.toggleTweetForm, __source: { fileName: _jsxFileName, lineNumber: 66 }, __self: this }, 'Cancel'),
            _react2.default.createElement(_reactBootstrap.Button, { id: 'submitForm', disabled: !this.isValidForm(), onClick: this.uploadPhotoBeforeCreatingTweet, __source: { fileName: _jsxFileName, lineNumber: 67 }, __self: this }, 'Tweet'))));




    }

    return component;
  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewTweetForm
    currentUser={
        Object {
            "all_tweet_count": 2,
            "followers": 1,
            "following": 1,
            "id": 1,
            "name": "Ryan",
            "retweet_count": 1,
            "slug": "rg",
            "token": "12345",
            "tweet_count": 1,
            "username": "RG",
          }
    }
    displayMessage={[Function]}
    toggleTweetForm={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <DarkLayer
          closeModal={[Function]}
/>,
        <div
          id="new_tweet_form"
>
          <div
                    className="modal_header"
          >
                    <h1>
                              Compose new Tweet
                    </h1>
          </div>
          <div
                    className="modal_body"
          >
                    <FormGroup
                              bsClass="form-group"
                    >
                              <FormControl
                                        bsClass="form-control"
                                        componentClass="textarea"
                                        id="form_body"
                                        maxLength="140"
                                        onChange={[Function]}
                                        placeholder="What's happening?"
                                        value=""
                              />
                    </FormGroup>
                    <FormGroup
                              bsClass="form-group"
                    >
                              <ControlLabel
                                        bsClass="control-label"
                                        htmlFor="photo_upload"
                                        srOnly={false}
                              >
                                        Add an image
                              </ControlLabel>
                              <FormControl
                                        bsClass="form-control"
                                        componentClass="input"
                                        id="photo_upload"
                                        type="file"
                              />
                    </FormGroup>
                    <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              disabled={false}
                              id="toggleForm"
                              onClick={[Function]}
                    >
                              Cancel
                    </Button>
                    <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              disabled={true}
                              id="submitForm"
                              onClick={[Function]}
                    >
                              Tweet
                    </Button>
          </div>
</div>,
      ],
      "id": "modal_wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "closeModal": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="modal_header"
>
              <h1>
                            Compose new Tweet
              </h1>
</div>,
            <div
              className="modal_body"
>
              <FormGroup
                            bsClass="form-group"
              >
                            <FormControl
                                          bsClass="form-control"
                                          componentClass="textarea"
                                          id="form_body"
                                          maxLength="140"
                                          onChange={[Function]}
                                          placeholder="What's happening?"
                                          value=""
                            />
              </FormGroup>
              <FormGroup
                            bsClass="form-group"
              >
                            <ControlLabel
                                          bsClass="control-label"
                                          htmlFor="photo_upload"
                                          srOnly={false}
                            >
                                          Add an image
                            </ControlLabel>
                            <FormControl
                                          bsClass="form-control"
                                          componentClass="input"
                                          id="photo_upload"
                                          type="file"
                            />
              </FormGroup>
              <Button
                            active={false}
                            block={false}
                            bsClass="btn"
                            bsStyle="default"
                            disabled={false}
                            id="toggleForm"
                            onClick={[Function]}
              >
                            Cancel
              </Button>
              <Button
                            active={false}
                            block={false}
                            bsClass="btn"
                            bsStyle="default"
                            disabled={true}
                            id="submitForm"
                            onClick={[Function]}
              >
                            Tweet
              </Button>
</div>,
          ],
          "id": "new_tweet_form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                Compose new Tweet
</h1>,
              "className": "modal_header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Compose new Tweet",
              },
              "ref": null,
              "rendered": "Compose new Tweet",
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FormGroup
                  bsClass="form-group"
>
                  <FormControl
                                    bsClass="form-control"
                                    componentClass="textarea"
                                    id="form_body"
                                    maxLength="140"
                                    onChange={[Function]}
                                    placeholder="What's happening?"
                                    value=""
                  />
</FormGroup>,
                <FormGroup
                  bsClass="form-group"
>
                  <ControlLabel
                                    bsClass="control-label"
                                    htmlFor="photo_upload"
                                    srOnly={false}
                  >
                                    Add an image
                  </ControlLabel>
                  <FormControl
                                    bsClass="form-control"
                                    componentClass="input"
                                    id="photo_upload"
                                    type="file"
                  />
</FormGroup>,
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  id="toggleForm"
                  onClick={[Function]}
>
                  Cancel
</Button>,
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={true}
                  id="submitForm"
                  onClick={[Function]}
>
                  Tweet
</Button>,
              ],
              "className": "modal_body",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-group",
                  "children": <FormControl
                    bsClass="form-control"
                    componentClass="textarea"
                    id="form_body"
                    maxLength="140"
                    onChange={[Function]}
                    placeholder="What's happening?"
                    value=""
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "componentClass": "textarea",
                    "id": "form_body",
                    "maxLength": "140",
                    "onChange": [Function],
                    "placeholder": "What's happening?",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-group",
                  "children": Array [
                    <ControlLabel
                      bsClass="control-label"
                      htmlFor="photo_upload"
                      srOnly={false}
>
                      Add an image
</ControlLabel>,
                    <FormControl
                      bsClass="form-control"
                      componentClass="input"
                      id="photo_upload"
                      type="file"
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "control-label",
                      "children": "Add an image",
                      "htmlFor": "photo_upload",
                      "srOnly": false,
                    },
                    "ref": null,
                    "rendered": "Add an image",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "form-control",
                      "componentClass": "input",
                      "id": "photo_upload",
                      "type": "file",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": false,
                  "bsClass": "btn",
                  "bsStyle": "default",
                  "children": "Cancel",
                  "disabled": false,
                  "id": "toggleForm",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Cancel",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": false,
                  "bsClass": "btn",
                  "bsStyle": "default",
                  "children": "Tweet",
                  "disabled": true,
                  "id": "submitForm",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Tweet",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DarkLayer
            closeModal={[Function]}
/>,
          <div
            id="new_tweet_form"
>
            <div
                        className="modal_header"
            >
                        <h1>
                                    Compose new Tweet
                        </h1>
            </div>
            <div
                        className="modal_body"
            >
                        <FormGroup
                                    bsClass="form-group"
                        >
                                    <FormControl
                                                bsClass="form-control"
                                                componentClass="textarea"
                                                id="form_body"
                                                maxLength="140"
                                                onChange={[Function]}
                                                placeholder="What's happening?"
                                                value=""
                                    />
                        </FormGroup>
                        <FormGroup
                                    bsClass="form-group"
                        >
                                    <ControlLabel
                                                bsClass="control-label"
                                                htmlFor="photo_upload"
                                                srOnly={false}
                                    >
                                                Add an image
                                    </ControlLabel>
                                    <FormControl
                                                bsClass="form-control"
                                                componentClass="input"
                                                id="photo_upload"
                                                type="file"
                                    />
                        </FormGroup>
                        <Button
                                    active={false}
                                    block={false}
                                    bsClass="btn"
                                    bsStyle="default"
                                    disabled={false}
                                    id="toggleForm"
                                    onClick={[Function]}
                        >
                                    Cancel
                        </Button>
                        <Button
                                    active={false}
                                    block={false}
                                    bsClass="btn"
                                    bsStyle="default"
                                    disabled={true}
                                    id="submitForm"
                                    onClick={[Function]}
                        >
                                    Tweet
                        </Button>
            </div>
</div>,
        ],
        "id": "modal_wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "closeModal": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="modal_header"
>
                <h1>
                                Compose new Tweet
                </h1>
</div>,
              <div
                className="modal_body"
>
                <FormGroup
                                bsClass="form-group"
                >
                                <FormControl
                                                bsClass="form-control"
                                                componentClass="textarea"
                                                id="form_body"
                                                maxLength="140"
                                                onChange={[Function]}
                                                placeholder="What's happening?"
                                                value=""
                                />
                </FormGroup>
                <FormGroup
                                bsClass="form-group"
                >
                                <ControlLabel
                                                bsClass="control-label"
                                                htmlFor="photo_upload"
                                                srOnly={false}
                                >
                                                Add an image
                                </ControlLabel>
                                <FormControl
                                                bsClass="form-control"
                                                componentClass="input"
                                                id="photo_upload"
                                                type="file"
                                />
                </FormGroup>
                <Button
                                active={false}
                                block={false}
                                bsClass="btn"
                                bsStyle="default"
                                disabled={false}
                                id="toggleForm"
                                onClick={[Function]}
                >
                                Cancel
                </Button>
                <Button
                                active={false}
                                block={false}
                                bsClass="btn"
                                bsStyle="default"
                                disabled={true}
                                id="submitForm"
                                onClick={[Function]}
                >
                                Tweet
                </Button>
</div>,
            ],
            "id": "new_tweet_form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                  Compose new Tweet
</h1>,
                "className": "modal_header",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Compose new Tweet",
                },
                "ref": null,
                "rendered": "Compose new Tweet",
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FormGroup
                    bsClass="form-group"
>
                    <FormControl
                                        bsClass="form-control"
                                        componentClass="textarea"
                                        id="form_body"
                                        maxLength="140"
                                        onChange={[Function]}
                                        placeholder="What's happening?"
                                        value=""
                    />
</FormGroup>,
                  <FormGroup
                    bsClass="form-group"
>
                    <ControlLabel
                                        bsClass="control-label"
                                        htmlFor="photo_upload"
                                        srOnly={false}
                    >
                                        Add an image
                    </ControlLabel>
                    <FormControl
                                        bsClass="form-control"
                                        componentClass="input"
                                        id="photo_upload"
                                        type="file"
                    />
</FormGroup>,
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={false}
                    id="toggleForm"
                    onClick={[Function]}
>
                    Cancel
</Button>,
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={true}
                    id="submitForm"
                    onClick={[Function]}
>
                    Tweet
</Button>,
                ],
                "className": "modal_body",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-group",
                    "children": <FormControl
                      bsClass="form-control"
                      componentClass="textarea"
                      id="form_body"
                      maxLength="140"
                      onChange={[Function]}
                      placeholder="What's happening?"
                      value=""
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "form-control",
                      "componentClass": "textarea",
                      "id": "form_body",
                      "maxLength": "140",
                      "onChange": [Function],
                      "placeholder": "What's happening?",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-group",
                    "children": Array [
                      <ControlLabel
                        bsClass="control-label"
                        htmlFor="photo_upload"
                        srOnly={false}
>
                        Add an image
</ControlLabel>,
                      <FormControl
                        bsClass="form-control"
                        componentClass="input"
                        id="photo_upload"
                        type="file"
/>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "bsClass": "control-label",
                        "children": "Add an image",
                        "htmlFor": "photo_upload",
                        "srOnly": false,
                      },
                      "ref": null,
                      "rendered": "Add an image",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "bsClass": "form-control",
                        "componentClass": "input",
                        "id": "photo_upload",
                        "type": "file",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "block": false,
                    "bsClass": "btn",
                    "bsStyle": "default",
                    "children": "Cancel",
                    "disabled": false,
                    "id": "toggleForm",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Cancel",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "block": false,
                    "bsClass": "btn",
                    "bsStyle": "default",
                    "children": "Tweet",
                    "disabled": true,
                    "id": "submitForm",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Tweet",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
